### PR TITLE
Issue #243

### PR DESCRIPTION
**Pull request by gregw**
Updated traceId format for TraceLoggingEnhancer for #243 

Signed-off-by: Greg Wilkins <gregw@webtide.com>

### CODE REVIEW

File: jetty9-base/src/main/java/com/google/cloud/runtimes/jetty9/RequestContextScope.java [hunk 1]
```diff
@@ -27,18 +27,21 @@
...
+ * @see TraceLoggingEnhancer
  */
 public class RequestContextScope implements ContextHandler.ContextScopeListener {
+
   static final Logger logger = Logger.getLogger(RequestContextScope.class.getName());
 
   private static final String X_CLOUD_TRACE = "x-cloud-trace-context";
   private static final ThreadLocal<Integer> contextDepth = new ThreadLocal<>();
 
+  private final String projectId = System.getProperty("app.deploy.project", "unknown");
```
Comment (meltsufin) [inline jetty9-base/src/main/java/com/google/cloud/runtimes/jetty9/RequestContextScope.java]: Who will set this system property?
I think you need to use `com.google.cloud.ServiceOptions.getDefaultProjectId()` instead.

Comment (gregw) [reply jetty9-base/src/main/java/com/google/cloud/runtimes/jetty9/RequestContextScope.java]: We have a few tests that do:
```
    datastore = DatastoreOptions.newBuilder()
        .setProjectId(System.getProperty("app.deploy.project"))
        .build()
        .getService();
```
Hence I assumed that property is available... but either way the ServiceOptions approach is much better.

**Comment by gregw** [comment]
@meltsufin I've been trying to test this with a [custom image](https://cloud.google.com/appengine/docs/flexible/custom-runtimes/build), but the results suggest that just stock images are being used? Any suggestions of how I can test this now?

File: jetty9-base/src/main/java/com/google/cloud/runtimes/jetty9/RequestContextScope.java (hunk 1 above)
Comment (meltsufin) [reply jetty9-base/src/main/java/com/google/cloud/runtimes/jetty9/RequestContextScope.java]: That was just designed for testing and it won't work normally. [See readme](https://github.com/GoogleCloudPlatform/jetty-runtime/blob/74a8e127bb7478b1927758a6115b9f8eb775348d/tests/README.md#remote-testing).

`com.google.cloud.ServiceOptions.getDefaultProjectId()` is the way to go.

**Comment by meltsufin** [comment]
It should work with the custom `Dockerfile`. I would check the build logs in Cloud Build. It should indicate which image was picked up and why.

**Comment by gregw** [comment]
@meltsufin I've not been able to work out how to deploy a custom image to test this.  I am following these [instructions](https://cloud.google.com/appengine/docs/flexible/java/configuring-your-app-with-app-yaml) against [this project](https://github.com/webtide/gae-dump)  with structure:
```
├── LICENSE-eplv10-aslv20.html
├── README.md
├── gae-dump.iml
├── pom.xml
├── src
│   └── main
│       ├── appengine
│       │   └── app.yaml
│       ├── docker
│       │   └── Dockerfile
│       ├── java
│       │   └── com
│       │       ├── example
│       │       │   └── appengine
│       │       └── webtide
│       │           └── gae
│       │               ├── ClassLoaderDump.java
│       │               ├── CookieDump.java
│       │               ├── Dump.java
│       │               └── SessionDump.java
│       └── webapp
│           ├── WEB-INF
│           │   ├── appengine-web.xml
│           │   └── web.xml
...
```
I'm trying to deploy using version 1.3.2 of the appengine-maven-plugin and the command:
```
mvn appengine:deploy
```
The deploy succeeds, but never uses the project app.yaml (even when moved to the top level) and always generates an app.yaml using java8 image rather than custom.

I've got a fresh install of gcloud sdk:
```
Google Cloud SDK 230.0.0
app-engine-java 1.9.71
app-engine-python 1.9.81
bq 2.0.39
cloud-datastore-emulator 2.0.3
core 2019.01.11
gsutil 4.35
```
running with 
```
java version "1.8.0_171"
Java(TM) SE Runtime Environment (build 1.8.0_171-b11)
Java HotSpot(TM) 64-Bit Server VM (build 25.171-b11, mixed mode)
```

**Comment by gregw** [comment]
@meltsufin doh! missed that I had both.  I appears to work fine with the just the `app.yaml`.  The documentation appears to favour that over `appengine-web.xml`, so I'll stick with that for now.
Good news is that I can see I'm formatting the trace tag correctly.... bad news is that it is not being used in the log scope?   Anyway, I can debug that now that I can run. thanks!

**Comment by gregw** [comment]
Appears to be [working now](https://console.cloud.google.com/logs/viewer?project=jetty9-work&authuser=2&resource=gae_app%2Fmodule_id%2Fdefault%2Fversion_id%2F20190119t153052&minLogLevel=0&expandAll=false&timestamp=2019-01-19T04:48:30.655000000Z&customFacets=&limitCustomFacetWidth=true&dateRangeStart=2019-01-19T03:48:30.905Z&dateRangeEnd=2019-01-19T04:48:30.905Z&interval=PT1H&logName=projects%2Fjetty9-work%2Flogs%2Fappengine.googleapis.com%252Fstdout&logName=projects%2Fjetty9-work%2Flogs%2Fappengine.googleapis.com%252Fstderr&logName=projects%2Fjetty9-work%2Flogs%2Fappengine.googleapis.com%252Fnginx.request&scrollTimestamp=2019-01-19T04:47:28.278000000Z)

**Pull request merged**